sdk/servicebus/service-bus/src/serializers/queueResourceSerializer.ts

```diff
@@ -255,6 +250,44 @@ export interface QueueProperties {
   forwardDeadLetteredMessagesTo?: string;
 }
 
+/**
+ * Fields that are updatable in the CreateQueueOptions.
+ */
+export type UpdatableFieldsForQueue =
+  | "defaultMessageTtl"
+  | "lockDuration"
+  | "deadLetteringOnMessageExpiration"
+  | "duplicateDetectionHistoryTimeWindow"
+  | "maxDeliveryCount";
+
+/**
+ * Fields that can assume undefined values in the QueueResponse.
+ */
+export type AllowUndefinedFieldsForQueue =
+  | "forwardTo"
+  | "authorizationRules"
+  | "forwardDeadLetteredMessagesTo";
+
+/**
+ * Represents the input for updateQueue.
+ *
+ * @export
+ * @interface QueueProperties
+ * @extends {Required<Pick<CreateQueueOptions, UpdatableFieldsForQueue>>}
+ * @extends {Readonly<Pick<CreateQueueOptions, AllowUndefinedFieldsForQueue>>}
+ * @extends {(Readonly<Required<Omit<CreateQueueOptions, UpdatableFieldsForQueue | AllowUndefinedFieldsForQueue>>>)}
+ */
+export interface QueueProperties
+  extends Required<Pick<CreateQueueOptions, UpdatableFieldsForQueue>>,
+    Readonly<Pick<CreateQueueOptions, AllowUndefinedFieldsForQueue>>,
+    Readonly<
+      Required<Omit<CreateQueueOptions, UpdatableFieldsForQueue | AllowUndefinedFieldsForQueue>>
+    > {
+  /**
+   * Status of the messaging entity.
+   */
+  status: EntityStatus;
```

Comment:
**Status param discrepancy**

https://docs.microsoft.com/en-us/rest/api/servicebus/create-queue

![image](https://user-images.githubusercontent.com/10452642/87844661-1dcda780-c874-11ea-923c-e18f95ab8483.png)


In JS, we have the following values for EntityStatus.

```ts
export type EntityStatus =
  | "Active"
  | "Creating"
  | "Deleting"
  | "ReceiveDisabled"
  | "SendDisabled"
  | "Disabled"
  | "Renaming"
  | "Restoring"
  | "Unknown";
```
**Questions**
- Why are the other possible values not mentioned in the [create-queue](https://docs.microsoft.com/en-us/rest/api/servicebus/create-queue) docs?
- From this - "Active: by default after creation, or when user activates it after last time suspended." It looks like Status can be updated after creating the queue, which makes it an updatable property but it is not mentioned in the docs for [update-queue](https://docs.microsoft.com/en-us/rest/api/servicebus/update-queue)

Reply:
From my testing, there appear to be several other properties that can be updated after a queue is created. Actually, the only ones that could **not** be updated were:
- RequiresSession
- RequiresDuplicateDetection
- EnablePartitioning
- Name

Reply:
> It looks like Status can be updated after creating the queue, which makes it an updatable property but it is not mentioned in the docs

My understanding of the docs above is that it is updatable by the service, not the user

Reply:
Status can be updated by the user.

Reply:
We also have tests that update the "Status".

Reply:
Can we have an issue logged to update the docs for create queue as well?

Reply:
Logged https://github.com/MicrosoftDocs/azure-docs/issues/59650